geospaas_harvesting/verify.py

```diff
@@ -0,0 +1,40 @@
+""" Verification module. """
+import os
+import sys
+from datetime import datetime
+
+import django
+import requests
+
+os.environ.setdefault('DJANGO_SETTINGS_MODULE', 'geospaas_harvesting.settings')
+django.setup()
+
+from geospaas.catalog.models import DatasetURI
+
+def main(filename):
+    """ Verifies the datasets based on their dataseturi. If the download link does not provide a
+    download availability and returns a text or html response, then the dataset uri is removed. If
+    there is no other "dataseturi" remains for the dataset, then the dataset is also removed in
+    order to be harvested again in the future with a correct and healthy "dataseturi".
+    Since the number of datasets in the database might be enormous, the datasets are retrieved
+    into with a variable named retrieved_dataset_uris with an specific length number
+    for memory management. """
+    init_index = 0
+    interval = 1000
+    with open(filename+".txt", 'w') as f:
+        while init_index < DatasetURI.objects.count():
+            try:
+                retrieved_dataset_uris = DatasetURI.objects.all()[init_index:init_index+interval]
+            except IndexError:
+                retrieved_dataset_uris = DatasetURI.objects.all()[init_index:]
+            init_index += interval
+            for dsuri in retrieved_dataset_uris:
+                content_type = requests.head(
+                    dsuri.uri, allow_redirects=True).headers.get('content-type')
+                if 'html' in content_type.lower() or 'text' in content_type.lower():
+                    f.write(dsuri.uri + os.linesep)
+
+
+if __name__ == '__main__':
+    main(filename=sys.argv[1] if len(sys.argv) == 2 else \
+            f"unverified_dataset_at_{datetime.now().strftime('%Y-%m-%d___%H_%M_%S')}")
```

Comment:
The code to get the file name get be put in the `main()` function since it's read from the command line.
It will be easier to write it in a clear and clean way there, rather than in this long line.